pkg/virt-handler/node-labeller/node_labeller_test.go

```diff
@@ -181,6 +205,62 @@ var _ = Describe("Node-labeller ", func() {
 		Expect(res).To(BeTrue())
 	})
 
+	DescribeTable("should re-enqueue node if its updating its", func(updateNode func(node *v1.Node)) {
+		emptyQueue()
+
+		updatedNode := addedNode.DeepCopy()
+		updateNode(updatedNode)
+		modifyNode(updatedNode)
+
+		Eventually(func() int {
+			return nlController.queue.Len()
+		}, 5*time.Second, time.Second).Should(Equal(1), "queue is expected to with length of 1")
+	},
+		Entry("labels", func(node *v1.Node) { node.Labels["new-key"] = "new-value" }),
+		Entry("annotations", func(node *v1.Node) { node.Annotations["new-key"] = "new-value" }),
+	)
+
+	It("should not re-enqueue node if it updated something other than labels / annotations", func() {
+		emptyQueue()
+
+		updatedNode := addedNode.DeepCopy()
+		updatedNode.Spec.PodCIDR = "fake"
+		mockQueue.ExpectAdds(1)
+		nodeSource.Modify(updatedNode)
+
+		Consistently(func() int {
+			return nlController.queue.Len()
+		}, 3*time.Second, time.Second).Should(Equal(0), "queue is expected to be empty")
```

Comment:
nit:
i think that`BeEmpty()` is better .

Reply:
I thought about that, but the problem is that `BeEmpty()` does not operate on ints as indicated by its documentation:
```
Actual must be of type string, array, map, chan, or slice.
```

We don't actually have a slice here, just an integer, that's why it's not possible. In any case, I've included a description to make the error understandable.